README.md
# :potable_water: Symbol Faucet


## :handshake: Using with catapult-service-bootstrap

### Build or Pull image

```console
# build image
$ docker build -t symbol-faucet .

# or pull from dockerhub
$ docker pull symbolplatform/symbol-faucet:0.4.0
```

### Add as service

#### Using specific PrivateKey and GenerationHash

```yaml:docker-compose.yml
faucet:
  image: symbolplatform/symbol-faucet:0.4.0
  stop_signal: SIGINT
  environment:
    - DEFAULT_NODE=http://rest-gateway:3000
    - FAUCET_PRIVATE_KEY=__YOUR_PRIVATE_KEY__
  ports:
    - '4000:4000'
  depends_on:
    - rest-gateway
```

#### Specific Mosaic faucet example

```yaml:docker-compose.yml
faucet:
  image: symbolplatform/symbol-faucet:0.4.0
  stop_signal: SIGINT
  environment:
    - DEFAULT_NODE=http://rest-gateway:3000
    - NATIVE_CURRENCY_ID=3E70742C9A38ACAB
  ports:
    - '4000:4000'
  depends_on:
    - rest-gateway
```

## :sparkles: Deploy to Heroku

[![Deploy](https://www.herokucdn.com/deploy/button.svg)](https://heroku.com/deploy)

Need to set `SYMBOL_PRIVATE_KEY`(PrivateKey of your faucet account) while deployment.

If you want to use ReCaptcha, set both variables `RECAPTCHA_CLIENT_SECRET` and `RECAPTCHA_SERVER_SECRET`.

## :whale: Dockerimage

- [symbolplatform\/symbol-faucet | Docker Hub](https://hub.docker.com/r/symbolplatform/symbol-faucet)

## :shell: Claimimg without Browser

```shell
curl http://localhost:4000/claims -d 'recipient=__YOUR_ADDRESS__'
```

## :fire: Customize

```shell
# set enviroment variables
# DEFAULT_NODE
# HOST
# PORT= (default: 3000)
# NATIVE_CURRENCY_NAME
# NATIVE_CURRENCY_ID
# NATIVE_CURRENCY_OUT_MAX
# NATIVE_CURRENCY_OUT_MIN
# FAUCET_PRIVATE_KEY
# MAX_FEE
# ENOUGH_BALANCE
# MAX_UNCONFIRMED
# BLACKLIST_MOSAIC_IDS
# EXPLORER_URL
# see .env.sample

# install packages
$ npm install

# start app
$ npm start

# or for development
$ npm run dev
```

## :muscle: Powered by

- [symbol-sdk-typescript-javascript](https://github.com/symbol/symbol-sdk-typescript-javascript)
- [nuxt/nuxt\.js: The Vue\.js Framework](https://github.com/nuxt/nuxt.js)
- [44uk/symbol\-faucet: Faucet application for symbol \(catapult\)](https://github.com/44uk/symbol-faucet)



CHANGELOG.md
# CHANGELOG
All notable changes to this project will be documented in this file.

The changelog format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).

## [1.1.0] - 31-Oct-2021

### Milestone: [catapult-server@v1.0.2.0](https://github.com/nemtech/catapult-server/releases/tag/v1.0.2.0)

Package  | Version  | Link
---|---|---
REST Core| v2.3.7 | [catapult-rest](https://github.com/nemtech/catapult-rest/releases/tag/v2.3.7)
SDK Core| v1.0.2 | [symbol-sdk](https://www.npmjs.com/package/symbol-sdk)

### Added
- Enable https  [#79](https://github.com/symbol/symbol-faucet/pull/79)
- Get node list from statistics service [#81](https://github.com/symbol/symbol-faucet/pull/81)

### Fixed
- Removed hardcoded XYM to support private networks [#76](https://github.com/symbol/symbol-faucet/pull/76)

### Update
- Symbol-sdk v1.0.2 [#78](https://github.com/symbol/symbol-faucet/pull/78)
- Readme [#80](https://github.com/symbol/symbol-faucet/pull/80)

## [1.0.0] - 14-Mar-2021

### Milestone: [catapult-server@v1.0.0.0](https://github.com/nemtech/catapult-server/releases/tag/v1.0.0.0)

Package  | Version  | Link
---|---|---
REST Core| v2.3.4 | [catapult-rest](https://github.com/nemtech/catapult-rest/releases/tag/v2.3.4)
SDK Core| v1.0.0 | [symbol-sdk](https://www.npmjs.com/package/symbol-sdk)

- Mainnet launch.

## [0.5.0] - 16-Jan-2021

### Milestone: [catapult-server@v0.10.0.5](https://github.com/nemtech/catapult-server/releases/tag/v0.10.0.5)

Package  | Version  | Link
---|---|---
REST Core| v2.3.0 | [catapult-rest](https://github.com/nemtech/catapult-rest/releases/tag/v2.3.0)
SDK Core| v0.23.0 | [symbol-sdk](https://www.npmjs.com/package/symbol-sdk)

### Added
- Added compatibility for 0.10.0.5 server.
- Added some message in homepage.

## [0.4.1] - 9-Dec-2020

### Milestone: [catapult-server@v0.10.0.4](https://github.com/nemtech/catapult-server/releases/tag/v0.10.0.4)

Package  | Version  | Link
---|---|---
REST Core| v2.2.0 | [catapult-rest](https://github.com/nemtech/catapult-rest/releases/tag/v2.2.0)
SDK Core| v0.22.0 | [symbol-sdk](https://www.npmjs.com/package/symbol-sdk)

### Added
- Added compatibility for 0.10.0.4 server.

### Fixed
- Lazy node health check when loading the page

## [0.4.0] - 24-Nov-2020

### Milestone: [catapult-server@v0.10.0.3](https://github.com/nemtech/catapult-server/releases/tag/v0.10.0.3)

Package  | Version  | Link
---|---|---
REST Core| v2.1.0 | [catapult-rest](https://github.com/nemtech/catapult-rest/releases/tag/v2.1.0)
SDK Core| v0.21.0 | [symbol-sdk](https://www.npmjs.com/package/symbol-sdk)

### Added
- improvement on toast notification.
- added error message hanlding for api node down or faucet init error.

## [0.3.0] - 26-Sep-2020

### Milestone: [catapult-server@v0.10.0.3](https://github.com/nemtech/catapult-server/releases/tag/v0.10.0.3)

Package  | Version  | Link
---|---|---
REST Core| v2.1.0 | [catapult-rest](https://github.com/nemtech/catapult-rest/releases/tag/v2.1.0)
SDK Core| v0.21.0 | [symbol-sdk](https://www.npmjs.com/package/symbol-sdk)

### Added
- Added compatibility for 0.10.0.3 server.


## [0.2.0] - 15-Aug-2020

### Milestone: [catapult-server@v0.9.6.4](https://github.com/nemtech/catapult-server/releases/tag/v0.9.6.4)

Package  | Version  | Link
---|---|---
REST Core| v1.2.0 | [catapult-rest](https://github.com/nemtech/catapult-rest/releases/tag/v1.2.0)
SDK Core| v0.20.7 | [symbol-sdk](https://www.npmjs.com/package/symbol-sdk)

### Added
- Added compatibility for 0.9.6.4 server.

### Update
- Enhanced UI/UX experience.


## [0.1.0] - 14-Apr-2020

### Milestone: [catapult-server@v0.9.3.2](https://github.com/nemtech/catapult-server/releases/tag/v0.9.3.2)


 Package  | Version  | Link
---|---|---
symbol-sdk| v0.17.3 | https://www.npmjs.com/package/symbol-sdk

- Added rebrand to Symbol.
- Added symbol icon, images, and color.
- Changed the footer to match the explorer.


[catapult-server@v1.0.0.0]: https://github.com/nemtech/catapult-server/releases/tag/v1.0.0.0

[catapult-rest@v2.3.4]: https://github.com/nemtech/catapult-rest/releases/tag/v2.3.4

server/config.ts
import helper from './helper';

const {
    DEFAULT_NODE,
    DEFAULT_NODE_CLIENT,
    NATIVE_CURRENCY_NAME,
    NATIVE_CURRENCY_ID,
    FAUCET_PRIVATE_KEY,
    NATIVE_CURRENCY_OUT_MAX,
    NATIVE_CURRENCY_OUT_MIN,
    MAX_FEE,
    ENOUGH_BALANCE,
    MAX_UNCONFIRMED,
    BLACKLIST_MOSAIC_IDS,
    EXPLORER_URL,
    STATISTICS_SERVICE_URL,
} = process.env;

export interface IConfig {
    DEFAULT_NODE: string;
    DEFAULT_NODE_CLIENT: string;
    NATIVE_CURRENCY_NAME: string;
    NATIVE_CURRENCY_ID: string;
    NATIVE_CURRENCY_OUT_MIN: number;
    NATIVE_CURRENCY_OUT_MAX: number;
    MAX_FEE: number;
    MAX_UNCONFIRMED: number;
    MAX_BALANCE: number;
    FAUCET_PRIVATE_KEY: string;
    BLACKLIST_MOSAICIDS: string[];
    EXPLORER_URL: string;
    STATISTICS_SERVICE_URL: string;
}

export const Config: IConfig = {
    DEFAULT_NODE: DEFAULT_NODE || 'http://localhost:3000',
    DEFAULT_NODE_CLIENT: DEFAULT_NODE_CLIENT || DEFAULT_NODE || 'http://127.0.0.1:3000',
    NATIVE_CURRENCY_NAME: NATIVE_CURRENCY_NAME || 'cat.currency',
    NATIVE_CURRENCY_ID: NATIVE_CURRENCY_ID || '06CCF23D2F3204EE',
    FAUCET_PRIVATE_KEY: FAUCET_PRIVATE_KEY || '9C8B1DEFDB700E8DA2834346E613E98D4C610093892B7E1AA39288C93709A462',
    NATIVE_CURRENCY_OUT_MIN: parseInt(NATIVE_CURRENCY_OUT_MIN || '') || 100000000,
    NATIVE_CURRENCY_OUT_MAX: parseInt(NATIVE_CURRENCY_OUT_MAX || '') || 500000000,
    MAX_FEE: parseFloat(MAX_FEE || '0') || 0,
    MAX_UNCONFIRMED: parseInt(MAX_UNCONFIRMED || '') || 99,
    MAX_BALANCE: parseInt(ENOUGH_BALANCE || '') || 100000000000,
    BLACKLIST_MOSAICIDS: helper.stringToArray(BLACKLIST_MOSAIC_IDS),
    EXPLORER_URL: EXPLORER_URL || 'http://localhost:80',
    STATISTICS_SERVICE_URL: STATISTICS_SERVICE_URL || '',
};
